EventFilter/SiPixelRawToDigi/plugins/SiPixelRawToDigiGPU.cc

```diff
@@ -373,19 +420,59 @@ SiPixelRawToDigiGPU::produce( edm::Event& ev, const edm::EventSetup& es)
     break;
   }
 
+  int32_t nclus=-1;
+  std::vector<AccretionCluster> aclusters(256);
+  auto totCluseFilled=0;
+  auto fillClusters = [&](uint32_t detId){
+    if (nclus<0) return; // this in reality should never happen
+    edmNew::DetSetVector<SiPixelCluster>::FastFiller spc(*outputClusters, detId);
+    auto layer = (DetId(detId).subdetId()==1) ? tTopo_->pxbLayer(detId) : 0;
+    auto clusterThreshold = (layer==1) ? 2000 : 4000;
+    for (int32_t ic=0; ic<nclus+1;++ic) {
+      auto const & acluster = aclusters[ic];
+      if ( acluster.charge < clusterThreshold) continue;
+      SiPixelCluster cluster(acluster.isize,acluster.adc, acluster.x,acluster.y, acluster.xmin,acluster.ymin);
+      ++totCluseFilled;
+      // std::cout << "putting in this cluster " << ic << " " << cluster.charge() << " " << cluster.pixelADC().size() << endl;
+      // sort by row (x)
+      spc.push_back( std::move(cluster) );
+      std::push_heap(spc.begin(),spc.end(),[](SiPixelCluster const & cl1,SiPixelCluster const & cl2) { return cl1.minPixelRow() < cl2.minPixelRow();});
+    }
+    // sort by row (x)
+    std::sort_heap(spc.begin(),spc.end(),[](SiPixelCluster const & cl1,SiPixelCluster const & cl2) { return cl1.minPixelRow() < cl2.minPixelRow();});
+    if ( spc.empty() ) spc.abort();
+   };
+
   for (uint32_t i = 0; i < wordCounterGPU; i++) {
     if (pdigi_h[i]==0) continue;
     assert(rawIdArr_h[i] > 109999);
     if ( (*detDigis).detId() != rawIdArr_h[i])
     {
+      fillClusters((*detDigis).detId());
+      nclus=-1; aclusters.clear();aclusters.resize(256);
```

Comment:
actually this is the time-waisting, we may just "clear" each acluster  (size and charge to 0)